docs/components/fab.md

```diff
@@ -0,0 +1,634 @@
+<!-- catalog-only-start --><!-- ---
+name: Floating Action Button (FAB)
+dirname: fab
+-----><!-- catalog-only-end -->
+
+<catalog-component-header>
+<catalog-component-header-title slot="title">
+
+# Floating Action Buttons (FAB)
+
+<!--*
+# Document freshness: For more information, see go/fresh-source.
+freshness: { owner: 'emarquez' reviewed: '2023-06-15' }
+tag: 'docType:reference'
+*-->
+
+<!-- go/md-fab -->
+<!-- [TOC] -->
+<!-- github-only-start -->
+
+TODO: update link to live site
+
+**This documentation is fully rendered on the
+[Material web catalog](https://github.com/material-components/material-web/tree/main)<!-- {.external} -->.**
+
+<!-- github-only-end -->
+
+[The FAB](https://m3.material.io/components/floating-action-button)<!-- {.external} -->
+represents the most important action on a screen. It puts key actions within
+reach
+
+[Extended FABs](https://m3.material.io/components/extended-fab) help people take
+primary actions. They're wider than FABs to accommodate a text label and larger
+target area.
+
+</catalog-component-header-title>
+
+<img
+    class="hero"
+    alt="A phone showing a payment screen with a green filled button that says 'Make
+payment'"
+    title="There are 5 types of common buttons"
+    src="images/fab/hero.webp">
+
+</catalog-component-header>
+
+*   Design articles
+    *   [FAB](https://m3.material.io/components/floating-action-button)
+        <!-- {.external} -->
+    *   [Extended FAB](https://m3.material.io/components/extended-fab)
+        <!-- {.external} -->
+*   API Documentation (*coming soon*)
+*   [Source code](https://github.com/material-components/material-web/tree/main/fab)
+    <!-- {.external} -->
+
+<!-- catalog-only-start -->
+
+<!--
+
+## Interactive Demo
+
+{% playgroundexample dirname=dirname %}
+
+-->
+
+<!-- catalog-only-end -->
+
+## Types
+
+### FAB
+
+<!-- github-only-start -->
+
+![The 3 sizes of FAB](images/fab/fabs.webp "There are three sizes of floating action buttons: FAB, small FAB, and large FAB")
+
+<!-- github-only-end -->
+<!-- catalog-include "figures/fab/fabs.html" -->
+
+1.  [FAB](#fab-1)
+2.  [Small FAB](#small)
+3.  [Large FAB](#large)
+
+### Extended FAB
+
+<!-- github-only-start -->
+
+![2 examples of extended FABs](images/fab/extended-fabs.webp "There are two types of extended FABs with and without icon")
+
+<!-- github-only-end -->
+<!-- catalog-include "figures/fab/extended-fabs.html" -->
+
+1.  [Extended FAB with both icon and label text](#with-icon)
+2.  [Extended FAB without icon](#without-icon)
+
+## Usage
+
+### FAB
+
+FAB comes in two variants, icon-only and extended (icon with label). There are
+also three sub variants to each, surface, primary, secondary, tertiary, and
```

Comment:
I got confused here. There are "three sub variants", but then 5 variants seem to be listed (surface, primary, secondary, tertiary, branded FAB). Or do each of these 5 have three sub variants?